### ISSUE TITLE
Olarm Home Assistant integration

### ISSUE THREAD

**Issue by boereworswizard** [parautenbach/Home-Automation#2]
I was using rainepretorius' integration which no longer works and stumbled across your method. Does it still work?

**Comment by parautenbach** [comment]
Yes, mine works (for me, at least). :-) I use it every day – including the automations. 

Shout if you find an issue.

**Comment by boereworswizard** [comment]
> Yes, mine works (for me, at least). :-) I use it every day – including the automations.
> 
> Shout if you find an issue.

Thanks for responding. I guess I must be doing something wrong!

Question 1: What do I do with the webhook secret created in Olarm?

Question 2: For the secrets.yaml, do I replace device_id with the P2****** device number listed on the Olarm app / website?

**Comment by boereworswizard** [comment]
I am trying to get this working now. I've setup the olarm.yaml and secret.yaml, added it to the configuration.yaml

However, it does not seem to be functioning correctly. Not syncing state and can't change state. 

Here are the error logs I am receiving:

```
Logger: homeassistant.components.automation.sync_alarm_state
Source: components/automation/__init__.py:763
integration: Automation (documentation, issues)
First occurred: 11:10:44 AM (1 occurrences)
Last logged: 11:10:44 AM

Error while executing automation automation.sync_alarm_state: Error rendering service name template: UndefinedError: 'str object' has no attribute 'deviceState'
```

```
Logger: homeassistant.components.automation.sync_alarm_state
Source: helpers/script.py:2026
integration: Automation (documentation, issues)
First occurred: 11:10:44 AM (1 occurrences)
Last logged: 11:10:44 AM

Sync Alarm State: Error executing script. Error for call_service at pos 3: Error rendering service name template: UndefinedError: 'str object' has no attribute 'deviceState'
```

```
Logger: homeassistant.helpers.template
Source: helpers/template.py:2742
First occurred: 11:10:44 AM (1 occurrences)
Last logged: 11:10:44 AM

Template variable error: 'str object' has no attribute 'deviceState' when rendering 'alarm_control_panel.alarm_{{ state_map[olarm_response.content.deviceState.areas[0]] }}'
```

```
Logger: homeassistant.components.rest_command
Source: components/rest_command/__init__.py:154
integration: RESTful Command (documentation, issues)
First occurred: 11:10:44 AM (1 occurrences)
Last logged: 11:10:44 AM

Error. Url: https://apiv4.olarm.co/api/v4/devices/RETRACTED-1dc0-4368-b15e-RETRACTED. Status code 403. Payload: None
```

**Comment by parautenbach** [comment]
Time for me to improve the instructions. :-) Thanks for using this and the feedback. 

Do you have all of the following defined in your `secrets.yaml`?

```
olarm_webhook_id: "REDACTED"
olarm_bearer_token: "Bearer api_REDACTED"
olarm_device_endpoint: "https://apiv4.olarm.co/api/v4/devices/REDACTED"
olarm_actions_endpoint: "https://apiv4.olarm.co/api/v4/devices/REDACTED/actions"
alarm_code: "REDACTED"
```

Are you using an externally accessible URL to your HA instance? If it's via Nabu Casa, then you webhook (on the Olarm side) must look something like this:

```
https://REDACTED.ui.nabu.casa/api/webhook/REDACTED
```

Where the second REDACTED is your `olarm_webhook_id`. 

> Status code 403.

This means "forbidden". 

Try doing a manual `curl` request from a command line and test your webhook from Olarm's UI.

**Comment by boereworswizard** [comment]
> Time for me to improve the instructions. :-) Thanks for using this and the feedback.
> 
> Do you have all of the following defined in your `secrets.yaml`?
> 
> ```
> olarm_webhook_id: "REDACTED"
> olarm_bearer_token: "Bearer api_REDACTED"
> olarm_device_endpoint: "https://apiv4.olarm.co/api/v4/devices/REDACTED"
> olarm_actions_endpoint: "https://apiv4.olarm.co/api/v4/devices/REDACTED/actions"
> alarm_code: "REDACTED"
> ```
> 
> Are you using an externally accessible URL to your HA instance? If it's via Nabu Casa, then you webhook (on the Olarm side) must look something like this:
> 
> ```
> https://REDACTED.ui.nabu.casa/api/webhook/REDACTED
> ```
> 
> Where the second REDACTED is your `olarm_webhook_id`.
> 
> > Status code 403.
> 
> This means "forbidden".
> 
> Try doing a manual `curl` request from a command line and test your webhook from Olarm's UI.

Thanks for attempting to help. I think some proper instructions would be great since it seems raine's integration no longer works and has been abandoned.

I think you might be on the right track why my integration doesn't work.... the webhook was indeed set to homeassistant.local which I have changed to my nabu.casa URL...

Furthermore, I am not sure if I created the webhook correctly. Do I use the "Receive Olarm Event (automation)" found under Settings > Cloud > Webhooks? Or do I create one under automations?

For olarm_webhook_id: Does it just need the ID, or the entire URL?

Appreciate you replying on a weekend!

**Comment by parautenbach** [comment]
> Furthermore, I am not sure if I created the webhook correctly. Do I use the "Receive Olarm Event (automation)" found under Settings > Cloud > Webhooks? Or do I create one under automations?

I'm not 100% sure that I follow you here, but you'll use the "Receive Olarm Event" automation. Are you configuring this via the UI?

> For olarm_webhook_id: Does it just need the ID, or the entire URL?

Just the ID.

**Comment by parautenbach** [comment]
I'm committing some instructions in the package YAML file, but I see I did include some instructions here: https://community.home-assistant.io/t/olarm-yaml-package/721266.
